test/e2e/utils_test.go

```diff
@@ -320,3 +320,28 @@ func getKubernetesLogs(t *testing.T, env environments.Environment, namespace, na
 	}
 	return string(out), nil
 }
+
+// httpGetResponseContains returns true if the response of GETting the URL contains specified substring.
+func httpGetResponseContains(t *testing.T, url string, client *http.Client, substring string) bool {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Log("failed to create request:",err)
+		return false
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Log("failed to get response:",err)
+		return false
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Log("failed to read response body:",err)
+		return false
+	}
+	
+	return strings.Contains(string(body), substring)
```

Comment:
I'd log the body here in case it doesn't contain the substring. This way we'll know what was the response and it will be easier to debug.

Reply:
I'd not log the body since it could produce lots of logs, considering its usage in our case:
- In our case, the response body is the whole Prometheus metrics, which would produce thousands of lines of logs in each `Log` call (1641 lines of body in my recent local run).
- it is called in `Never` or `Eventually` loops, so it prints log in each iteration of loop, produces even more logs.

Reply:
How about sending such things to a "log dump" and really log it down when something goes wrong.

Reply:
This sounds great **but** the problem might be to find yourself in that log dump. Imagine you send lots of logs from different places, different objects, tests etc to a single "log dump".

If you're going through a full test suite log dump then you have the context, you know where a test started, failed and ended.